File: M08_Architecture/C02_Component/S03_ViewBinding/src/main/java/net/bi4vmr/study/useinadapter/MyAdapter.java
package net.bi4vmr.study.useinadapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import net.bi4vmr.study.R;
import net.bi4vmr.study.databinding.ListItemSimpleBinding;

import java.util.List;

/**
 * Name        : MyAdapter
 * <p>
 * Author      : BI4VMR
 * <p>
 * Email       : bi4vmr@outlook.com
 * <p>
 * Date        : 2023-04-04 15:38
 * <p>
 * Description : RecyclerView的适配器。
 */
public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    // 上下文环境
    private final Context mContext;
    // 数据源
    private final List<SimpleVO> dataSource;

    public MyAdapter(Context context, List<SimpleVO> dataSource) {
        this.mContext = context.getApplicationContext();
        this.dataSource = dataSource;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(mContext);
        // 获取Item布局文件对应的视图绑定类实例
        ListItemSimpleBinding itemBinding = ListItemSimpleBinding.inflate(inflater, parent, false);
        // 创建ViewHolder实例，并传入视图绑定对象。
        return new MyViewHolder(itemBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.bindData();
    }

    @Override
    public int getItemCount() {
        return dataSource.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {

        // 只需要保存视图绑定对象即可，不需要再保存每个控件的引用。
        private final ListItemSimpleBinding binding;

        public MyViewHolder(@NonNull ListItemSimpleBinding binding) {
            super(binding.getRoot());
            // 保存视图绑定对象
            this.binding = binding;
        }

        public void bindData() {
            // 获取当前项的数据
            SimpleVO vo = dataSource.get(getAdapterPosition());
            // 将数据设置到控件中
            binding.tvTitle.setText(vo.getTitle());
        }
    }
}

File: M01_Overview/C00_Temporary/src/main/kotlin/net/bi4vmr/study/MyAdapter.kt
package net.bi4vmr.study

import android.view.LayoutInflater
import android.view.ViewGroup
import net.bi4vmr.study.databinding.ListItemType1Binding

/**
 * TODO 添加简述。
 *
 * TODO 添加详情。
 *
 * @author bi4vmr@outlook.com
 * @since 1.0.0
 */
class MyAdapter : BindingAdapter<VO>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BindingViewHolder<*, VO> {
        val binding = ListItemType1Binding.inflate(LayoutInflater.from(parent.context), parent, false)
        return VH(binding)
    }
}

data class VO(
    val title: String,
    val iconRes: Int
) : ListItem

class VH(binding: ListItemType1Binding) : BindingViewHolder<ListItemType1Binding, VO>(binding) {

    override fun bindData(i: VO) {
        binding.tvTitle.text = i.title
        binding.ivIcon.setImageResource( i.iconRes)
    }
}
